docs/docs/getting_started/usage-and-installation.mdx

```diff
@@ -0,0 +1,734 @@
+
+```
+yarn
+```
+
+3. Install the Python dependencies:
+
+```
+pip install -r requirements.txt
+```
+
+If you are using a M1 chipped Mac and getting a `'gmp.h' file not found` error when installing Cairo run the following:
+
+```
+CFLAGS=-Ibrew --prefix gmp/include LDFLAGS=-Lbrew --prefix gmp/lib pip install ecdsa fastecdsa sympy
+```
+
+Then run the pip command above again.
+
+4. Compile the project:
+
+```
+yarn tsc
+yarn warplib
+```
+
+5. Test the installation worked by transpiling an example ERC20 contract:
+
+```
+bin/warp transpile example_contracts/ERC20.sol
+```
+
+## Usage
+
+Warp 2 allows transpiling directly from Solidity to Cairo and transpiled .cairo file can be deployed to starknet.
+
+### Transpiling
+
+To transpile a solidity file, run:
+
+```
+warp transpile [PATH_TO_SOLIDITY_FILE]
+```
+
+For example:
+
+```
+warp transpile contracts/token/ERC20/ERC20.sol
+```
+
+The output of the transpilation could be found in warp_output folder.
+
+Now,the file can be compiled and deployed. Run the following to compile the Cairo contract:
+
+```
+warp compile warp_output/contracts/token/ERC20/ERC20__WARP_CONTRACT__ERC20.cairo
+```
+
+### Deploying
+
+```
+warp deploy [PATH_TO_TRANSPILED_CAIRO_CONTRACT] --use_cairo_abi --inputs  [INPUT_ARGUMENTS]
+```
+
+For example:
+
+```
+warp deploy warp_output/contracts/token/ERC20/ERC20__WARP_CONTRACT__ERC20.cairo --use_cairo_abi --inputs  10,0x4e,0x45,0x54,0x48,0x45,0x52,0x43,0x4f,0x49,0x4e,4,0x4e,0x45,0x54,0x48
+```
+
+#### Note
+
+In the example above, we needed a name and ticker for our ERC20 coin. We used NETHERCOIN and NETH.
+StarkNet does not support Solidity-style strings,
+so the Warp team treats them as dynamic arrays of bytes.
+To match the StarkNet abi we need to pass the length of the arrays and the value of each byte.
+When using --use_cairo_abi, one thing to take care of is it only supports negative numbers in 2's complement form.
+
+### A Detailed Example
+
+Let us get hands-on with the transpiling and deploying process starting with a solidity file using warp cli commands.
+
+We have a solidity smart contract file named `simple_deployment_example.sol` with code :
+
+```
+pragma solidity ^0.8;
+
+contract TestFunction1 {
+  uint256 public x;
+  uint256 public y;
+  string public name;
+  uint256[] public z;
+
+  constructor(
+    uint256 _x,
+    uint256 _y,
+    string memory _name,
+    uint256[] memory _z
+  ) {
+    x = _x;
+    y = _y;
+    name = _name;
+    z = _z;
+  }
+
+  function setVal(
+    uint256 _x,
+    uint256 _y,
+    string memory _name,
+    uint256[] memory _z
+  ) public {
+    x = _x;
+    y = _y;
+    name = _name;
+    z = _z;
+  }
+}
+```
+
+Now, to transpile it to cairo, we need to run the following command :
+
+```
+warp transpile simple_deployment_example.sol
+```
+
+The output file(.cairo) which you can find in warp_output folder looks like :
+
+```
+%lang starknet
+
+from starkware.cairo.common.dict import dict_read, dict_write
```

Comment:
use small output example

Reply:
The solidity file is small, with only one constructor and a invoke function, and both take arguments of various types as uint, array, and string. It is kept such to let users know how to pass different types of arguments via the warp CLI command. But the transpiled output came longer.